src/Stratis.Features.Unity3dApi/NFTTransferIndexer.cs

```diff
@@ -0,0 +1,239 @@
+
+                        ChainedHeader chainTip = this.chainIndexer.Tip;
+
+                        List<ReceiptResponse> receipts = this.smartContractTransactionService.ReceiptSearch(
+                            contractAddr, "TransferLog", null, currentContract.LastUpdatedBlock, null);
+
+                        if (receipts == null)
+                            continue;
+
+                        int lastReceiptHeight = 0;
+                        if (receipts.Any())
+                            lastReceiptHeight = (int)receipts.Last().BlockNumber.Value;
+
+                        currentContract.LastUpdatedBlock = new List<int>() { chainTip.Height, lastReceiptHeight }.Max();
+
+                        List<TransferLog> transferLogs = new List<TransferLog>(receipts.Count);
+
+                        foreach (ReceiptResponse receiptRes in receipts)
+                        {
+                            string jsonLog = Newtonsoft.Json.JsonConvert.SerializeObject(receiptRes.Logs.First().Log);
+                            
+                            TransferLog infoObj = JsonConvert.DeserializeObject<TransferLog>(jsonLog);
+                            transferLogs.Add(infoObj);
+                        }
+                    
+                        foreach (TransferLog transferInfo in transferLogs)
+                        {
+                            if (currentContract.OwnedIDsByAddress.ContainsKey(transferInfo.From))
+                            {
+                                currentContract.OwnedIDsByAddress[transferInfo.From].Remove(transferInfo.TokenId);
+
+                                if (currentContract.OwnedIDsByAddress[transferInfo.From].Count == 0)
+                                    currentContract.OwnedIDsByAddress.Remove(transferInfo.From);
+                            }
+
+                            if (!currentContract.OwnedIDsByAddress.ContainsKey(transferInfo.To))
+                                currentContract.OwnedIDsByAddress.Add(transferInfo.To, new List<long>());
+
+                            currentContract.OwnedIDsByAddress[transferInfo.To].Add(transferInfo.TokenId);
+                        }
+
+                        this.NFTContractCollection.Upsert(currentContract);
+                    }
+
+                    try
+                    {
+                        await Task.Delay(TimeSpan.FromSeconds(6), this.cancellation.Token);
+                    }
+                    catch (TaskCanceledException)
+                    {
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError(e.ToString());
+            }
+        }
+
+        public void Dispose()
+        {
+            this.cancellation.Cancel();
+            this.indexingTask?.GetAwaiter().GetResult();
+            this.db.Dispose();
```

Comment:
The value of `this.db` may be null if Initialize had not been called.

Reply:
nftIndexer.Dispose is not called if this.apiSettings.EnableUnityAPI is false. 
but added ? anyway